woocommerce/payment-gateway/class-sv-wc-payment-gateway-payment-form.php

```diff
@@ -1013,7 +1032,27 @@ public function render_fieldset_end() {
 	 */
 	public function render_js() {
 
-		wc_enqueue_js( $this->get_safe_handler_js() );
+		// skip if script already rendered
+		if ( true === $this->js_rendered ) {
+			return;
+		}
```

Comment:
I wonder if this simple memoization could hinder problems in the rare occurrence there are multiple gateways initializing different scripts - I know it's unlikely, but maybe the cache key should account for the gateway ID e.g. with `$this->get_gateway()->get_id()`

Reply:
do you mean something like a static array property holds a list of gateway IDs of script rendered? ex:
```php
static $js_rendered = [];

public function render_js() {
    if ( in_array( $this->get_gateway()->get_id(), static::$js_rendered, true ) ) {
        returnl
    }
}
```

Reply:
something like that, don't think needs to be static though